task/sql-tasks.js

```diff
@@ -280,30 +461,49 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+  let result = await db.query(`
+        SELECT DISTINCT
+            C.CompanyName,
+            P.ProductName,
+            OD.UnitPrice as "PricePerItem"
+        FROM Customers as C
+        JOIN Orders as O ON O.CustomerID = C.CustomerID        
+        JOIN OrderDetails as OD ON OD.OrderID = O.OrderID
+        JOIN Products as P ON OD.ProductID = P.ProductID
+        JOIN (SELECT
+          CNew.CustomerID,
+          MAX(ODNew.UnitPrice) as 'MaxPrice'
+          FROM Customers as CNew
+          JOIN Orders as ONew ON ONew.CustomerID = CNew.CustomerID 
+          JOIN OrderDetails as ODNew ON ONew.OrderID = ODNew.OrderID 
+          GROUP BY CNew.CustomerID) as S
+          ON S.CustomerID = C.CustomerID AND S.MaxPrice = OD.UnitPrice
+          ORDER BY OD.UnitPrice DESC, C.CompanyName, P.ProductName;
+  `);
+  return result[0];
 }
 
 module.exports = {
-    task_1_1: task_1_1,
-    task_1_2: task_1_2,
-    task_1_3: task_1_3,
-    task_1_4: task_1_4,

```

Comment:
do not modify original code

Reply:
i don`t understand. i didn`t modify anything